src/docs_website/src/file_checker.zig

```diff
@@ -70,5 +70,62 @@ fn check_files(arena: std.mem.Allocator, path: []const u8) !void {
                 return error.MissingNewline;
             }
         }
+
+        if (std.mem.eql(u8, file_type.extension, ".html")) {
+            const html_path = try std.fs.path.join(arena, &.{ path, entry.path });
+            try check_links(arena, html_path);
+        }
+    }
+}
+
+// These links don't work with https.
+const https_exceptions = std.StaticStringMap(void).initComptime(.{
+    .{"http://www.bailis.org/blog/linearizability-versus-serializability/"},
+});
+
+fn check_links(arena: std.mem.Allocator, html_path: []const u8) !void {
+    const dirname = std.fs.path.dirname(html_path).?;
+    const html = try std.fs.cwd().readFileAlloc(arena, html_path, file_size_max);
+
+    var line_it = std.mem.tokenizeScalar(u8, html, '\n');
+    var line_number: usize = 1;
+    while (line_it.next()) |line| {
```

Comment:
This is a bit hard to read, as it mixes together link extraction and link validation, both of which are fiddly logic. 

Let's try wringing it as

```zig
var link_iterator = find_links(html);
while (link_iterator.next_link()) |link| {

}
```

where 

```zig
const link = struct {
    line_number: u32,
    uri: std.Uri,
};
```

maybe even extracting logic for validating a single link into a helper function, to keep "loop over links" control plane and "check one link" data plane separate.